### PR TITLE
[framework] dont null assert in _debugVerifyIllFatedPopulation

### PR DESCRIPTION
**Pull request by jonahwilliams**
If you run the tests in profile mode, this will fail due to the null check. Since we don't explicitly disallow profile mode in tests, I would prefer to keep it working - and if we decided to stop supporting it that can be an intention rather than accidental decision

### CODE REVIEW

**Comment by jonahwilliams** [comment]
@Hixie for lack of tests....

**Review by chandarrengoog — APPROVED** [review]

File: packages/flutter/lib/src/widgets/framework.dart [hunk 1]
```diff
@@ -2879,7 +2879,7 @@ class BuildOwner {
   void _debugVerifyIllFatedPopulation() {
     assert(() {
       Map<GlobalKey, Set<Element>>? duplicates;
-      for (final Element element in _debugIllFatedElements!) {
+      for (final Element element in _debugIllFatedElements ?? <Element>{}) {
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/framework.dart:2882]: const?

**Comment by Hixie** [comment]
If you want profile builds to be testable, I strongly recommend actually testing it. Right now they only work by pure chance (and sometimes don't work because we assume e.g. that kDebugMode matches when assert() is called). The only reason they work at all right now is that Google's internal tests rely on it in a couple of places. Several times we've broken those tests by mistake. If those tests go away (and I think that's quite possible) then the next time we break it we may not notice.

**Comment by jonahwilliams** [comment]
Fair enough - I think the plan might be (@dnfield ) to remove support for profile mode with asserts. I think the challenge with testing this configuration is that we may not have any tooling in flutter/flutter that supports building profile with asserts enabled. (at least not since I last looked at the tool..)

File: packages/flutter/lib/src/widgets/framework.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter/lib/src/widgets/framework.dart]: done

**Comment by dnfield** [comment]
I filed https://github.com/flutter/flutter/issues/96552 to track

**Comment by jonahwilliams** [comment]
I could write a devicelab test that asserts that it works, then @dnfield could later change it to assert failure.

**Comment by jonahwilliams** [comment]
actually I don't need devicelab, I can use a similar hack to test_release. tests added

**Comment by jonahwilliams** [comment]
Without the change to framework.dart, the test fails with:

```
PS C:\Users\Jonah\flutter\packages\flutter> flutter test --dart-define=dart.vm.profile=true .\test_profile\
00:05 +0: Can build widget tree in profile mode with asserts enabled
══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
════════════════════════════════════════════════════════════════════════════════════════════════════
══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
════════════════════════════════════════════════════════════════════════════════════════════════════
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
════════════════════════════════════════════════════════════════════════════════════════════════════
00:05 +0 -1: Can build widget tree in profile mode with asserts enabled [E]
  Test failed. See exception logs above.
  The test description was: Can build widget tree in profile mode with asserts enabled

```

**Review by dnfield — COMMENTED** [review]
LGTM until we can have a better way to enforce what modes unit tests are run in.

**Comment by jonahwilliams** [comment]
Mac_android run_release_test is hitting an infra failure, known issue?

**Comment by dnfield** [comment]
@CaseyHillers or @keyonghan might know (https://ci.chromium.org/ui/p/flutter/builders/try/Mac_android%20run_release_test/18/overview). I triggered a rerun.

**Comment by CaseyHillers** [comment]
> @CaseyHillers or @keyonghan might know (https://ci.chromium.org/ui/p/flutter/builders/try/Mac_android%20run_release_test/18/overview). I triggered a rerun.

I filed https://github.com/flutter/flutter/issues/96672 to the ticket queue. If you don't need Mac_android coverage, feel free to ack it and submit it to see what CI says. I'll look into getting a few extra devices added to the try pool so one device doesn't control presubmit.

**Comment by jonahwilliams** [comment]
cool, thank you @CaseyHillers !

**Pull request merged**